sympy/solvers/ode.py

```diff
@@ -1452,6 +1460,40 @@ def _test_term(coeff, order):
     else:
         return tuple(retlist)
 
+def separate_vars(f, symbols=[]):
+
+    from sympy import ordered
+
+    if symbols:
+        if not all((t.is_Atom for t in symbols)):
+            raise ValueError("symbols must be Atoms.")
+    elif symbols is None:
+        return {'coeff':f}
+    else:
+        symbols = list(ordered(f.free_symbols))
+        if not symbols:
+            return None
+
+    f = simplify(f)
+    expsym = f.free_symbols
+    intersection = list(ordered(set(symbols).intersection(expsym)))
+    if len(intersection) < 2:
+        return _separatevars_dict(f, symbols)
+    elif len(intersection) == 2:
+         b = Dummy('b')
+         a = Dummy('a')
+         f1 = f.subs(intersection[1], b)
+         f2 = f.subs(intersection[0], a)
+         f3 = f1.subs(intersection[0], a)
+         f4 = expand(cancel((f1*f2)/f3))
+         f = expand(cancel(f))
+         if not any(i in simplify(expand(cancel(f1*f2/f))).free_symbols for i in symbols):
+             return _separatevars_dict(Mul(*[simplify(f1/f3),f2]),symbols)
```

Comment:
Don't forget to restore the original symbols; your answer is coming back with a and b in it.

Also, do you mean to send back all symbols or just intersection?

Reply:
> Also, do you mean to send back all symbols or just intersection?


all symbols